src/goto-instrument/cover.cpp

```diff
@@ -1461,6 +1461,31 @@ bool instrument_cover_goals(
       cmdline.isset("no-trivial-tests"));
   }
 
+  if(cmdline.isset("cover-traces-must-terminate"))
+  {
+    // instrument a 'fake' goal in CPROVER_START. This will rephrase
+    // the reachability problem  by asking BMC to provide a solution that
+    // satisfies a goal while getting to the end of the program-under-test.
+    const auto sf_it=
+      goto_functions.function_map.find(goto_functions.entry_point());
+    if(sf_it==goto_functions.function_map.end())
+    {
+      msg.error() << "cover-traces-must-terminate: invalid entry point"
+        << messaget::eom;
+      return true;
+    }
+    auto if_it=sf_it->second.body.instructions.end();
+    while(!if_it->is_function_call())
+      if_it--;
+    if_it++;
+    const std::string &comment="fake goal, reachability constraint.";
```

Comment:
I'd prefer not calling it "fake", but maybe "Additional goal to ensure complete trace coverage"